File: cpp/array/125. Valid Palindrome.cpp
#include <string>
#include <cctype>

class Solution {
public:
    bool isPalindrome(std::string s) {
        int left = 0, right = s.size()-1;

        while (left < right) {
            while (left < right && !std::isalnum(s[left])){
                left++;
            }
            while (left < right &&!std::isalnum(s[right])){
                right--;
            }
            if (std::tolower(s[left]) != std::tolower(s[right])){
                return false ;
            }
            left++;
            right--; 
        }
        return true; 
    }
};

File: php/125. Valid Palindrome.php
<?php

class Solution {

    function isPalindrome($s) {
        $s = preg_replace('/[^a-z0-9]/', '', strtolower($s));
        return $s === strrev($s);
    }

    /**
     * @param String $s
     * @return Boolean
     */
    function isPalindrome2($s) {
        $left = 0;
        $right = strlen($s)-1;

        while ($left < $right) {
            while($left < $right && !ctype_alnum($s[$left])) {
                $left++;
            }
            while($left < $right && !ctype_alnum($s[$right])) {
                $right--;
            }
            if ($left < $right && strtolower($s[$left]) !== strtolower($s[$right])) {
                return false; 
            }
            $left++;
            $right--;
        }
        
        return true;
    }
}

?>